xonsh/lexer.py

```diff
@@ -370,11 +384,13 @@ def get_tokens(s, tolerant):
     Given a string containing xonsh code, generates a stream of relevant PLY
     tokens using ``handle_token``.
     """
+    input_cp = io.BytesIO(s.encode("utf-8"))
+    del s
```

Comment:
Is there any reason to do this beyond a _very_ tiny memory savings?

Reply:
Yes that is the reasin. I was testing with some huge file. Do you want to remove ?

Reply:
It should get cleaned up shortly afterwards by garbage collection, no?

Reply:
I wasn't sure about that. It yields the dict and might hold reference to the locals. anyway, this is small size normally I will remove it. Let me know, if you want me to remove the LexToken class?